File: backend/app/api/routes/face_routes.ts
import { Router } from 'express';
import * as faceService from '../../services/face_service';

const router = Router();

router.post('/register-face', async (req, res) => {
  try {
    const { name, email, faceDescriptor } = req.body;
    if (!name || !email || !faceDescriptor) {
      return res.status(400).json({ error: 'Missing required fields' });
    }
    const user = await faceService.registerUser(name, email, faceDescriptor);
    res.status(201).json(user);
  } catch (error) {
    console.error('Registration error:', error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

router.post('/verify-face', async (req, res) => {
  try {
    const { faceDescriptor } = req.body;
    if (!faceDescriptor) {
      return res.status(400).json({ error: 'Missing face descriptor' });
    }
    const user = await faceService.verifyFace(faceDescriptor);
    if (user) {
      res.json({ success: true, user: { id: user.id, name: user.name, email: user.email } });
    } else {
      res.status(401).json({ success: false, error: 'Face not recognized' });
    }
  } catch (error) {
    console.error('Verification error:', error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

router.get('/users', async (req, res) => {
  try {
    const users = await faceService.getAllUsers();
    res.json(users);
  } catch (error) {
    res.status(500).json({ error: 'Internal Server Error' });
  }
});

export default router;

File: backend/app/api/routes/face_routes.py
from fastapi import APIRouter, HTTPException
from ...services import face_service
from ...schemas.face_schema import UserRegister, FaceVerify, UserResponse
from typing import List

router = APIRouter()

@router.post("/register-face", response_model=UserResponse)
async def register_face(user_data: UserRegister):
    try:
        user = await face_service.register_user(
            user_data.name, 
            user_data.email, 
            user_data.faceDescriptor
        )
        return user
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

@router.post("/verify-face")
async def verify_face(verify_data: FaceVerify):
    user = await face_service.verify_face(verify_data.faceDescriptor)
    if user:
        return {"success": True, "user": user}
    else:
        raise HTTPException(status_code=401, detail="Face not recognized")

@router.get("/users", response_model=List[UserResponse])
async def get_users():
    return await face_service.get_all_users()
